src/lib/ndpi_main.c

```diff
@@ -4502,6 +4521,65 @@ static int ndpi_handle_rule(struct ndpi_detection_module_struct *ndpi_str,
 
 /* ******************************************************************** */
 
+/*
+ * Format:
+ *
+ * <proto,param,value>
+ *
+ * Notes:
+ *  - proto might be empty
+ *  - empty lines or lines starting with # are ignored
+ */
+int load_config_file_fd(struct ndpi_detection_module_struct *ndpi_str, FILE *fd) {
+  char buffer[512], *line, *proto, *param = NULL, *value, *saveptr;
+  int len;
+  ndpi_cfg_error rc;
+
+  if(!ndpi_str || !fd)
+    return -1;
+
+  while(1) {
+    line = fgets(buffer, sizeof(buffer), fd);
+
+    if(line == NULL)
+      break;
+
+    len = strlen(line);
+
+    if((len <= 1) || (line[0] == '#'))
+      continue;
+
+    line[len - 1] = '\0';
+
+    if(line[0] == ',') { /* First parameter might be missing */
```

Comment:
That makes some config files/strings kinda ugly. Another possible way could be to use the return value of `strtok_r` to also count the amount of found `","`s.

Reply:
Yea, exactly. :D

Reply:
Ok, it is a good idea. I will try that in the next iteration

Reply:
Done